plutus-ledger-api/test-plugin/Spec/Data/ScriptContext.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE BangPatterns          #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NegativeLiterals      #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PatternSynonyms       #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE ViewPatterns          #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:context-level=0 #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:defer-errors #-}
+
+module Spec.Data.ScriptContext where
+
+import Test.Tasty (TestTree)
+import Test.Tasty.Extras
+
+import PlutusLedgerApi.Data.V3 qualified as V3D
+import PlutusTx.Builtins qualified as PlutusTx
+import PlutusTx.Code
+import PlutusTx.IsData qualified as PlutusTx
+import PlutusTx.Prelude qualified as PlutusTx
+import PlutusTx.Test
+import PlutusTx.TH (compile)
+
+tests :: TestTree
+tests =
+  runTestNested ["test-plugin", "Spec", "Data", "SriptContext"] . pure . testNestedGhc $
+      [ goldenPirReadable "alwaysSucceeds" compiledAlwaysSucceeds
+      , goldenUPlcReadable "alwaysSucceeds" compiledAlwaysSucceeds
+      , goldenPirReadable "succeedsIfHasDatum" compiledSucceedsIfHasDatum
+      , goldenUPlcReadable "succeedsIfHasDatum" compiledSucceedsIfHasDatum
+      ]
+
+alwaysSucceeds :: PlutusTx.BuiltinData -> PlutusTx.BuiltinUnit
+alwaysSucceeds d =
+    PlutusTx.check $
```

Comment:
Since this is a new code I expect it to be fourmolu-formatted. 
However, an indentation is inconsistent: sometimes its 2 spaces, sometimes 4.